### DISCUSSION TITLE
Quick develop WB and Tone Control

### DISCUSSION THREAD

**Discussion by nadavrotem** [rsjaffe/MIDI2LR#555] [Ideas]
Hi,
I cant believe that using the Quick develop WB and Tone Control feature is still not addressed by Adobe. It`s a crime against photographers!
I don`t want/need to go into the slow Develop Module!
I saw a post somewhere with this nice guy trying to answer Jaffe request from Adobe :
Add to API: LrDevelopController.quickDevelopAdjustImage smallDecrease and largeDecrease

He wrote:

> Some reverse engineering reveals how to use the photo:quickDevelop* methods to behave like the <<, <, >, and >> buttons in the Quick Develop panel.

photo:quickDevelopAdjustWhiteBalance ("Temperature", n)
photo:quickDevelopAdjustWhiteBalance ("Tint", n)

With n = -5 or 5, these settings increase or decrease the same amount as using the < and > buttons. With n = -20 and 20, they change the same amount as using the << and >> buttons.

photo:quickDevelopAdjustImage ("Exposure", n)

With n = -16.5 or 16.5, Exposure increases or decreases the same amount as using the < and > buttons. With n = -50 or 50, Exposure changes the same amount as using the << and >> buttons.

photo:quickDevelopAdjustImage (setting, n)

For "setting" equal to one of "Contrast", "Highlights", "Shadows", "Whites", "Blacks", "Clarity", and "Vibrance": With n = -5 or 5, the settings increase or decrease the same amount as using the < and > buttons. With n = -20 or 20, they change the same amount as using the << and >> buttons.

* * * 

With all of these, when the settings reach their minimum or maximum value, further attempts to decrease or increase them leave them at the minimum/maximum, just like the <<, <, >, and >> buttons.

And of course, this should all be documented.  " end of quote

Is this helpful? Can this feature be added finally ?
Please and Thank you

**Comment by rsjaffe** [comment]
I'll work on adding this to the next release.

**Reply by nadavrotem** [reply]
Thank you so much! 
That is from my point of view (and I think a lot more people will appreciate that) a major update :)

**Comment by rsjaffe** [comment]
If you want to test it out, Extract the three lua files from [ClientUtilities.zip](https://github.com/rsjaffe/MIDI2LR/files/7566057/ClientUtilities.zip), place them in your MIDI2LR.lrplugin directory, delete the _MenuTrans.xml_ file that is found in _%AppData%\MIDI2LR_ (Windows) or _~/Library/Application Support/MIDI2LR_ (MacOS) and start up Lightroom.

**Reply by nadavrotem** [reply]
Hi and thank you so much!! you are the best!
There is one important thing missing and one bug I found so far:
Missing - Temperature and Tint control in the White Balance tab
Bug - Increase shadows and decrease shadows (At -5 +5 increments) do the opposite

**Reply by rsjaffe** [reply]
1. Increase and decrease shadows use the same terminology that Lightroom uses. If you hover over the quick develop arrows for shadows, you'll see that **-5** is called **increase shadows**, because the lower the value is, the more shadows in the picture. I'd rather it was termed differently, but I'm following lightroom.
2. Temperature and tint should be in the test. Have you deleted the _MenuTrans.xml_ file? The added items won't show unless you both install the new lua files _and_ delete _MenuTrans.xml_. That forces MIDI2LR to regenerate its menus.

**Reply by nadavrotem** [reply]
1. ok :)
2. I don't see it and I followed your steps, should I do them again? if so in what order exactly? (attaching a screenshot of what I see)
![Screenshot_5](https://user-images.githubusercontent.com/6247427/142739211-eab7da29-29ef-4225-b6ab-dc8940766d1c.jpg)
3. I use the Behringer Xtouch Mini - Is it possible for MIDI2LR to recognize turning the knobs Right and Left as two different commands?
It will be a dream to use the knobs on the Library module, it has the potential of being a super fast workflow!
Thanks!

**Reply by rsjaffe** [reply]
For the patch to work, you need to
1. Copy the files into MIDI2LR.lrplugin directory where the main plugin is installed.
2. Delete the MenuTrans.xml file.
3. Restart Lightroom

**Reply by nadavrotem** [reply]
I am doing that but Tint and Temperature still dont show up :(

**Reply by rsjaffe** [reply]
Just wait until the new version comes out, then.

**Comment by rsjaffe** [comment]
It's in 5.2
